tests/integration/test_relation.py

```diff
@@ -103,17 +104,25 @@ async def test_delete_unused_ingresses(ops_test: OpsTest, app_name: str):
     assert isinstance(ops_test.model, Model)
     model_name = ops_test.model_name
 
-    def compare_svc_hostnames(expected: List[str]) -> bool:
-        all_ingresses = api_networking.list_namespaced_ingress(namespace=model_name)
-        return expected == [ingress.spec.rules[0].host for ingress in all_ingresses.items]
-
-    assert compare_svc_hostnames(["any"])
+    def assert_svc_hostnames(expected: List[str], timeout=300):
+        time_start = time.time()
+        while True:
+            all_ingresses = api_networking.list_namespaced_ingress(namespace=model_name)
+            try:
+                assert expected == [ingress.spec.rules[0].host for ingress in all_ingresses.items]
```

Comment:
There isn't an equivalent of list comprehension for tuple, you can use a constructor + generator expression, but I don't think there's much benefit in this situation since the list here is just compared once and discarded.

Reply:
Yes that is true, there is no tuple comprehension. Tuples are still slightly more efficient if it is a constant since the machinery for lists doesn't have to be created. And usually it is better to use tuples if it is a constant so you don't accidentally make changes. In this case, there is probably no practical difference. It is more than in these kind of cases there are benefits to systematically use tuples to reduce the chance of mistakes

Reply:
Okay, I update to tuple. But warning ahead, the single element tuple is kind ugly.